pkg/kubecost/etlset.go

```diff
@@ -0,0 +1,15 @@
+package kubecost
+
+import "encoding"
+
+// ETLSet is an interface which represents the basic data block of an ETL. It is keyed by its Window
+type ETLSet interface {
+	ConstructSet() ETLSet
+	CloneSet() ETLSet
```

Comment:
Any reason this shouldn't just be called `Clone()`?

Reply:
Yes, the Clone() signature is taken across the codebase with the return value of the impl type and cannot easily be updated to return `ETLSet`. On the other hand `CloneSet()` can be quickly implemented for classes that already have `Clone()` by just returning the `Clone()` which would not require a massive refactor.

Reply:
Got it! that makes sense.